app/controllers/billing_invoice.py

```diff
@@ -118,3 +121,64 @@ def post(self, project_id):
             message=f'Invoice for user with email {user.email} and project_id {project_id} sent successfully',
             data=dict(invoice=new_invoice_data)
         ), 200
+
+
+    @jwt_required
+    def get(self, project_id):
+
+        current_user_id = get_jwt_identity()
+        current_user_roles = get_jwt_claims()['roles']
+
+        billing_invoice_schema = BillingInvoiceSchema(many=True)
+        project = Project.get_by_id(project_id)
+
+        billing_invoice = BillingInvoice.find_all(project_id=project_id)
+
+        if not billing_invoice:
+            return dict(
+                status='fail',
+                message=f'billing invoice records not found'
+            ), 404
+        
+        
+        if not is_owner_or_admin(project, current_user_id, current_user_roles):
```

Comment:
This should come before getting the billing invoice. For consistency, if a user isn't an owner of the project, It should always return unauthorised before telling them the invoice is not found